Makefile

```diff
@@ -41,72 +35,29 @@ endif
 # Specify the date of build
 BUILD_DATE = $(shell date +'%Y%m%d%H%M%S')
 
-# Tools required for different make targets or for development
-EXTERNAL_TOOLS=\
-	gopkg.in/alecthomas/gometalinter.v1
-
-all: $(addprefix build-, $(BIN))
-
-build-%:
-	$(MAKE) --no-print-directory BIN=$* build
-
-build: _output/$(BIN)
-
-_output/$(BIN): $(BIN)/*.go
-	mkdir -p .go/src/$(REPO) .go/pkg .go/std/$(ARCH) _output
-	docker run \
-				 --rm \
-				 -u $$(id -u):$$(id -g) \
-				 -v $$(pwd)/.go/pkg:/go/pkg \
-				 -v $$(pwd)/.go/src:/go/src \
-				 -v $$(pwd)/.go/std:/go/std \
-				 -v $$(pwd):/go/src/$(REPO) \
-				 -v $$(pwd)/.go/std/$(ARCH):/usr/local/go/pkg/linux_$(ARCH)_static \
-				 -e CGO_ENABLED=0 \
-				 -w /go/src/$(REPO) \
-				 $(BUILD_IMAGE) \
-				 go build -installsuffix "static" -i -v -o _output/$(BIN) ./$(BIN)
+all: ci
 
 container: all
-	cp Dockerfile _output/Dockerfile
+	@cp Dockerfile _output/Dockerfile
 	docker build -t $(IMAGE):$(IMAGE_TAG) --build-arg BUILD_DATE=${BUILD_DATE} -f _output/Dockerfile _output
 
-all-ci: $(addprefix ci-, $(BIN))
-
-ci-%:
-	$(MAKE) --no-print-directory BIN=$* ci
-
 ci:
-	mkdir -p _output
+	@mkdir -p _output
 	CGO_ENABLED=0 go build -v -o _output/$(BIN) ./$(BIN)
 
-format:
-	@echo "--> Running go fmt"
-	@go fmt $(PACKAGES)
-
-# Bootstrap the build by downloading additional tools
-bootstrap:
-	@for tool in  $(EXTERNAL_TOOLS) ; do \
-		echo "+ Installing $$tool" ; \
-		go get -u $$tool; \
-	done
-
-vet:
-	go vet ${PACKAGES}
-
-# Target to run gometalinter in Travis (deadcode, golint, errcheck, unconvert, goconst)
-golint-travis:
-	@gometalinter.v1 --install
-	-gometalinter.v1 --config=metalinter.config ./...
-
-golint:
-	@gometalinter.v1 --install
-	@gometalinter.v1 --vendor --disable-all -E errcheck -E misspell ./...
+# Run linter using docker image
+lint-docker:
+	@docker run -i								\
+		--rm -v $$(pwd):/app -w /app 			\
+		golangci/golangci-lint:v1.24.0 			\
+		bash -c "GOGC=75 golangci-lint run"
 
-check: golint-travis format vet
+# Run linter using local binary
+lint:
+	@golangci-lint run
 
```

Comment:
We can also add the `deps` target to manage the vendor dependencies easily
```
.PHONY: deps
deps: ## Ensures fresh go.mod and go.sum.
	@go mod tidy
	@go mod verify
```

Reply:
when i ran go mod tidy, i got error
```
github.com/openebs/velero-plugin/pkg/clouduploader imports
	gocloud.dev/blob/gcsblob tested by
	gocloud.dev/blob/gcsblob.test imports
	gocloud.dev/internal/testing/setup imports
	github.com/Azure/azure-storage-blob-go/azblob tested by
	github.com/Azure/azure-storage-blob-go/azblob.test imports
	github.com/Azure/go-autorest/autorest/adal: ambiguous import: found package github.com/Azure/go-autorest/autorest/adal in multiple modules:
	github.com/Azure/go-autorest v12.0.0+incompatible (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/adal)
	github.com/Azure/go-autorest/autorest/adal v0.5.0 (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest/autorest/adal@v0.5.0)
github.com/openebs/velero-plugin/velero-blockstore-cstor imports
	github.com/vmware-tanzu/velero/pkg/plugin/framework tested by
	github.com/vmware-tanzu/velero/pkg/plugin/framework.test imports
	github.com/vmware-tanzu/velero/pkg/test imports
	github.com/vmware-tanzu/velero/pkg/client imports
	k8s.io/client-go/plugin/pkg/client/auth/azure imports
	github.com/Azure/go-autorest/autorest: ambiguous import: found package github.com/Azure/go-autorest/autorest in multiple modules:
	github.com/Azure/go-autorest v12.0.0+incompatible (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest)
	github.com/Azure/go-autorest/autorest v0.9.0 (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.0)
github.com/openebs/velero-plugin/velero-blockstore-cstor imports
	github.com/vmware-tanzu/velero/pkg/plugin/framework tested by
	github.com/vmware-tanzu/velero/pkg/plugin/framework.test imports
	github.com/vmware-tanzu/velero/pkg/test imports
	github.com/vmware-tanzu/velero/pkg/client imports
	k8s.io/client-go/plugin/pkg/client/auth/azure imports
	github.com/Azure/go-autorest/autorest/azure: ambiguous import: found package github.com/Azure/go-autorest/autorest/azure in multiple modules:
	github.com/Azure/go-autorest v12.0.0+incompatible (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest@v12.0.0+incompatible/autorest/azure)
	github.com/Azure/go-autorest/autorest v0.9.0 (/home/mayank/go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.0/azure)
```
but go mod verify return 0.
```
all modules verified
```
What could be the reason?

Reply:
it happened due to conflict. Importing ambiguous import package with `go get package@none`  fix the issue.